discovery-provider/src/utils/helpers.py

```diff
@@ -519,3 +521,27 @@ def get_solana_tx_owner(meta, idx) -> str:
         ),
         "",
     )
+
+
+def get_final_poa_block(shared_config):
+    # get final poa block from identity and cache result
+    # marks the transition to nethermind
+    # returns None if still on POA
+    redis = redis_connection.get_redis()
+    cached_final_poa_block = redis_get_or_restore(redis, final_poa_block_redis_key)
+    if cached_final_poa_block:
+        return cached_final_poa_block
+
+    identity_endpoint = (
+        f"{shared_config['discprov']['identity_service_url']}/health_check/poa"
+    )
+
+    response = requests.get(identity_endpoint, timeout=1)
+    response.raise_for_status()
```

Comment:
I don't think we gracefully handle exceptions being throw in this request, right? Should we

try:
  response.raise_for_status()
except:
  return None

Reply:
it would be pretty cool to send some kind of "tombstone" transaction to a special address that when the indexer sees it it knows to cut over... but might be complicated, so polling identity seems like an okay option.